torch/testing/_internal/common_methods_invocations.py

```diff
@@ -198,7 +198,7 @@ def __init__(self,
         self.dtypes = set(dtypes)
         self.dtypesIfCPU = set(dtypesIfCPU) if dtypesIfCPU is not None else self.dtypes
         self.dtypesIfCUDA = set(dtypesIfCUDA) if dtypesIfCUDA is not None else self.dtypes
-        self.dtypesIfROCM = set(dtypesIfROCM) if dtypesIfROCM is not None else self.dtypes
+        self.dtypesIfROCM = set(dtypesIfROCM) if dtypesIfROCM is not None else self.dtypesIfCUDA
```

Comment:
@jithunnair-amd i think this is more reasonable fix --> when dtypeIfROCM is missing. it should default to dtypesIfCUDA instead of falling back to the non GPU one IMO

Reply:
Yes, this does seem to be the correct way to do it, to ensure the flexibility of specifying a restricted set for ROCm if needed, but otherwise use the CUDA set.